Update OS updates guide (explain behavior when past deadline date is set) (#27492)
fleetdm/fleet: articles/enforce-os-updates.md

## End user experience

### Apple (macOS, iOS, and iPadOS)

On macOS hosts, when a minimum version is enforced, end users see a native macOS notification (DDM) once per day. Users can choose to update ahead of the deadline or schedule it for that night. 24 hours before the deadline, the notification appears hourly and ignores Do Not Disturb. One hour before the deadline, the notification appears every 30 minutes and then every 10 minutes.

> Certain user preferences may suppress macOS update notifications. To prevent users from being surprised by a forced update or unexpected restart, consider communicating OS update deadlines through additional channels.

On iOS and iPadOS hosts, end users will see a notification in their Notification Center after the deadline. They can’t use their iPhone or iPad until the OS update is installed.

If the host was turned off when the deadline passed, the update will be scheduled an hour after it’s turned on.

> If you set a past date (ex. yesterday) as the deadline, the end user will immediately be prompted to install the update. If they don't, the update will automatically install in one hour. Similarly, if you set the deadline to today, end users will experience the same behavior if it is still before 12 PM (end user local time).

Reviewer: @marko-lisica what happens if I set the deadline to today and it's _after_ 12 PM?

Author: @noahtalerman This is a mistake. I reversed it here. If it's after 12 PM, then it's considered past date, same behavior as if you set yesterdays date.

Gabe still needs to confirm what happens if it's **after 12 PM**. See this [thread](https://fleetdm.slack.com/archives/C03C41L5YEL/p1742842690065809).

Added a suggestion below, and we can add what happens if it's after 12 PM, once Gabe tests that.

> If you set a past date (ex. yesterday) as the deadline, the end user will immediately be prompted to install the update. If they don't, the update will automatically install in one hour. Similarly, if you set the deadline to today, end users will experience the same behavior if it is still before 12 PM (end user local time).

Author: ```suggestion
If you set a past date (ex. yesterday) as the deadline, the end user will immediately be prompted to install the update. If they don't, the update will automatically install in one hour. Similarly, if you set the deadline to today, end users will experience the same behavior if it's after 12 PM (end user local time).
```
